### ISSUE TITLE
Don't publish unneeded artefacts

### ISSUE THREAD

**Issue by xlotlu** [lektor/lektor#754]
Ideally artefacts that are unused should not be published, where unused == nothing links to them.

Since some people rely on publishing things regardless, it should be configurable, defaulting to false in the beginning and switching to true a few releases (and warnings) later.

Of much interest are images that are never supposed to be accessed directly, but only their "thumbnails" should be published. During building, the I/O / local storage benefits would largely be negated by #753, but it would save a lot on remote publishing, both in terms of speed and storage space.

**Comment by runfalk** [comment]
I think we do have to consider how that would be handled in the plugin API so we don't break plugins like https://github.com/fschulze/lektor-rst

I also think we should allow overriding this on a per file basis. I for instance put a robots.txt in my root record.

**Comment by xlotlu** [comment]
Yes, it will need some careful consideration. Maybe a glob pattern in settings. Ideally it could be set per item, that would mean abusing hidden fields in contents.lr, I can't think of any other way.

Anyway, right now it's wishful thinking for the distant future.

**Comment by runfalk** [comment]
I don't think glob pattern is good as the only solution since you can't change it from within the admin. It doesn't make sense that you can upload files but not force include them in the build. I think the file metadata feature is probably good usage for this. In case you are not aware it is possible to create a file with the same name as an attachment but with `.lr` suffix. There it's possible to change system fileds like slug and visibility.

**Comment by xlotlu** [comment]
I know, attachments can even have models.

So it should be:
 - glob in settings
 - glob in contents.lr
 - field in file.lr
